priam/src/main/java/com/netflix/priam/identity/InstanceIdentity.java

```diff
@@ -300,4 +308,40 @@ public void updateToken() throws Exception {
         myInstance.setToken(tokenManager.sanitizeToken(nodetool.getTokens().get(0)));
         instanceRegistry.update(myInstance);
     }
+
+    // filter other DC's
+    private List<PriamInstance> filteredRemote(List<PriamInstance> priamInstances) {
+        List<PriamInstance> local = Lists.newArrayList();
+        for (PriamInstance priamInstance : priamInstances) {
+            if (priamInstance.getLocation().equals(location)) {
+                local.add(priamInstance);
+            }
+        }
+        return local;
+    }
+
+    /**
+     * Simple class for maintaining the pair of (location, availability zone), which can be used for grouping
+     * instances in the same ring by availability zone.
+     */
+    private final class LocationAZPair {
+        final Location location;
+        final String availabilityZone;
+
+        public LocationAZPair(Location location, String availabilityZone) {
+            this.location = location;
+            this.availabilityZone = availabilityZone;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            LocationAZPair other = (LocationAZPair) o;
+            return location.equals(other.location) && availabilityZone.equals(other.availabilityZone);
```

Comment:
`LocationAZPair` is a private class whose use is limited and controlled, so the only circumstances where it is being compared with another instance is when it's another `LocationAZPair`, so adding that seemed like unnecessary bloat.  However, I can add it if only so that future caretakers long after we're gone don't ask the same question (or break something because it isn't there).